### DISCUSSION TITLE
REST API does not provide more than 1000 nodes(workflow runs) for GET request

### DISCUSSION THREAD

**Discussion by ayberkyasa** [community/community#41630] [Apps, API and Webhooks]
### Select Topic Area

Bug

### Body

Hi folks!

I have an issue when I try to fetch data from the following endpoint: /repos/{owner}/{repo}/actions/runs?branch=main&per_page=100. This endpoint is from https://docs.github.com/en/rest/actions/workflow-runs?apiVersion=2022-11-28#list-workflow-runs-for-a-repository.

When I try to use the above endpoint, I can access at most 10 pages. On the 10th page, there is a next-page link in the response header for the 11th page. However, when I try to go to the 11th page, the next-page link in the response header disappears and there is also no data on the 11th page. By the way, the last-page link in the response header is around 270. It is true because the repository from which I fetched workflow runs data has almost 27K workflow runs in their main branch.

If I delete the "branch=main" query parameter from the request endpoint, it works without the above problem. Moreover, If I write "per_page=50" instead of "per_page=100" in the request endpoint, I can access at most 20 pages. The same problem explained above started on the 21st page.

It may be wrong but I reach the conclusion that GitHub REST API does not provide more than 1000 nodes (workflow runs data) if I use the "branch=<branch_name>" query parameter in the request. However, I have to fetch all workflow runs data for the "main" branch. This seems to be impossible for now due to the above problem in GitHub REST API.

Is there anyone who can explain the underlying reason for this problem to me? Can GitHub resolve this problem?

**Comment by jelgun** [comment]
same issue (or feature?) on the GitHub web itself: e.g., [https://github.com/elastic/elasticsearch/actions?page=60&query=branch%3Amain+](https://github.com/elastic/elasticsearch/actions?page=60&query=branch%3Amain+)

**Comment by aranair** [comment]
Also hitting this with the official `octokit` sdks. 
- Anything with a `created` filter applied ontop will pull up to the 1000th record. (e.g. 10 pages with 100 per_page)
- If I remove that filter, and just let it pull everything, I can get every node

**Comment by CWebsterMS** [comment]
ah `This API will return up to 1,000 results for each search when using the following parameters: actor, branch, check_suite_id, created, event, head_sha, status.`
